<center> **Solving a Linear Programming Problem** </center>

In [1]:
from scipy.optimize import linprog
import numpy as np

Objective function:
$$z(max) = 5x_1 + 4x_2$$

Constraints:
$$C_1 = x_1 + x_2 \leq 5$$
$$C_2 = 10*x_1 + 6*x_2 \leq 45$$
$$x_1,x_2 \geq 0$$

In [2]:
z = np.array([5,4])

C for constraints, which is going to store all constraints of the problem.

In [3]:
C = np.array([
    [1, 1],          #C1
    [10,6]           #C2
])

Then store 5 and 45 for each constraint.

In [4]:
b = np.array([5,45])

We have to specify our bounds, in this case is $\geq0$ for each variable $x_1$ and $x_2$.

In [5]:
x1 = (0, None)
x2 = (0, None)

To solve the problem we need to call linprog, and store the solution in a variable. 

We are going to use sol.

In [6]:
sol = linprog(-z, A_ub = C, b_ub = b, bounds = (x1, x2), method='simplex')

In [7]:
print(sol)

     fun: -23.75
 message: 'Optimization terminated successfully.'
     nit: 2
   slack: array([ 0.,  0.])
  status: 0
 success: True
       x: array([ 3.75,  1.25])


To print the values that are needed we are going to use:

In [8]:
print("x1 = {}, x2 = {}, z = {}".format(sol.x[0], sol.x[1], sol.fun*-1))

x1 = 3.75, x2 = 1.25, z = 23.75


Our LP problem is solved now we are going to replace those values into the objective function.

$$z(max) = 5x_1 + 4x_2$$
$$z(max) = 5*3.75 + 4*1.25$$
$$z(max) = 23.75$$

Using an easy way by multiplying our z variable ```z = np.array([7,6])``` and our ```sol.x``` values.

In [9]:
OF = z*sol.x
print("z = {}".format(OF.sum()))

z = 23.75
